plugin-server/src/main/job-queues/job-queue-manager.ts

```diff
@@ -50,7 +58,35 @@ export class JobQueueManager implements JobQueue {
         }
     }
 
-    async enqueue(jobName: string, job: EnqueuedJob): Promise<void> {
+    async enqueue(jobName: string, job: EnqueuedJob, instrumentationContext?: InstrumentationContext): Promise<void> {
+        const jobType = 'type' in job ? job.type : 'buffer'
+        const jobPayload = 'payload' in job ? job.payload : job.eventPayload
+        const pluginServerMode = this.pluginsServer.PLUGIN_SERVER_MODE ?? 'full'
+        await instrument(
+            this.pluginsServer.statsd,
+            {
+                metricName: jobName === JobName.PLUGIN_JOB ? 'vm.enqueuePluginJob' : 'vm.enqueueBufferJob',
+                key: instrumentationContext?.key ?? '?',
+                tag: instrumentationContext?.tag ?? '?',
+                tags: { pluginServerMode, type: jobType },
+                data: { timestamp: job.timestamp, type: jobType, payload: jobPayload },
+            },
+            () =>
+                runRetriableFunction({
+                    hub: this.pluginsServer,
+                    metricPrefix: 'enqueueJob',
+                    metricName: jobName,
+                    metricTags: {
+                        pluginServerMode,
+                    },
+                    tryFn: async () => this._enqueue(jobName, job),
+                    catchFn: () => status.error('🔴', 'Exhausted attempts to enqueue job.'),
```

Comment:
Problems:
- The log message isn't actionable or even clear. Suggestion: `Exhausted attempts to enqueue job, job was dropped.`
- We should log the `jobName, job` objects as well

Q:
- This is a `catchFn` not `finallyFn`? Does this swallow all errors or will sentry receive an error as well? T